### DISCUSSION TITLE
Help importing bump library

### DISCUSSION THREAD

**Discussion by 12buntu** [nesbox/TIC-80#1293] [Q&A]
Perhaps this is a very stupid question, but I'm having trouble using bump.lua. The wiki has it listed under the "Libraries" page, however I can't get it to work. I have it in the same directory as the cartridge I'm trying to run, however it doesn't work for some reason. It says it can't find `./bump.lua` what am I doing wrong?

Relevant code: 
```lua
-- script: lua
local bump = require 'bump'
```
Error:
![image](https://user-images.githubusercontent.com/36310448/103449177-89fd6580-4c72-11eb-9b36-a12b75daa1c8.png)

**Comment by RobLoach** [comment]
You'll have to embed the library directly, since there is no file system. Lua Amalgamate is an option.

**Reply by 12buntu** [reply]
> You'll have to embed the library directly, since there is no file system. Lua Amalgamate is an option.

I can't seem to get Amalgamate to work, [this](https://github.com/siffiejoe/lua-amalg) is what you're talking about right?

I guess I'll just have to copy/paste the entire library into the top of my code.

EDIT: I'm dumb
EDIT2: Wait am I dumb? What happened to banks? I took a break from TIC-80 for a while and now the feature is gone, but the sync function still appears to do... something?

**Reply by RobLoach** [reply]
Yes, `lua-amalg` :wink: 

I've also worked on a small utility that concatenates all dependencies, while keeping the tree in-tact. It's using Node.js, but works with Lua scripts: https://github.com/robloach/scriptpacker

**Comment by nesbox** [comment]
Only code banks are gone a lot of problems with them, they actually never worked as banks because you can't `sync()` code.
So, now we have one big code bank again which can be compressed if > 64K :)

**Reply by 12buntu** [reply]
Okay I guess that makes sense, I did enjoy the code banks from an organizational perspective though.
Is there any "official" way to use libraries? I'm willing to write some documentation on it if there is a good way.

**Comment by Tyler887** [comment]
> Only code banks are gone a lot of problems with them, they actually never worked as banks because you can't `sync()` code. So, now we have one big code bank again which can be compressed if > 64K :)

@nesbox's answer above is right.

**Reply by nesbox** [reply]
Yes, it's right, we don't have code banks anymore, you only have 64K in the regular build (also can be compressed) and 64K*8 in PRO instead.
